File: examples/sample-java/src/main/java/com/monetai/sample/java/MainActivity.java
package com.monetai.sample.java;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.monetai.sample.java.databinding.ActivityMainBinding;
import com.monetai.sdk.MonetaiSDKJava;
import com.monetai.sdk.models.AppUserDiscount;
import com.monetai.sdk.models.PredictResult;
import com.monetai.sdk.models.PaywallConfig;
import com.monetai.sdk.models.PaywallStyle;
import com.monetai.sdk.models.Feature;
import com.monetai.sdk.models.ViewProductItemParams;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.Arrays;
import java.util.List;



public class MainActivity extends AppCompatActivity {
    
    private ActivityMainBinding binding;
    private static final String TAG = "MainActivity";
    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("MMM dd, HH:mm", Locale.getDefault());
    private final ExecutorService executor = Executors.newSingleThreadExecutor();
    
    // MARK: - Paywall State
    private PaywallConfig paywallConfig;
    
    // MARK: - Fake Products
    private FakeProductAdapter fakeProductAdapter;
    private final List<FakeProduct> fakeProducts = Arrays.asList(
        new FakeProduct(
            "fake_basic_monthly",
            "Basic Plan",
            "Essential features for starters",
            4.99,
            9.98,
            "USD",
            1
        ),
        new FakeProduct(
            "fake_pro_yearly",
            "Pro Plan",
            "Advanced tools for power users",
            59.99,
            119.98,
            "USD",
            12
        )
    );
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        
        setupUI();
        
        // Log app launch event BEFORE SDK initialization
        executor.execute(() -> MonetaiSDKJava.getShared().logEvent("app_in"));
        
        setupMonetaiSDK();
    }
    
    @Override
    protected void onDestroy() {
        super.onDestroy();
        executor.shutdown();
    }
    
    private void setupUI() {
        // Setup button click listeners
        binding.predictButton.setOnClickListener(v -> predictButtonTapped());
        binding.logEventButton.setOnClickListener(v -> logEventButtonTapped());
        
        // Fake products list
        fakeProductAdapter = new FakeProductAdapter(this::logFakeProductViewed);
        binding.recyclerViewFakeProducts.setLayoutManager(new LinearLayoutManager(this));
        binding.recyclerViewFakeProducts.setAdapter(fakeProductAdapter);
    }
    
    private void setupMonetaiSDK() {
        System.out.println("🎯 Setting up Monetai SDK...");
        
        MonetaiSDKJava.getShared().initialize(
            this,
            Constants.SDK_KEY, // SDK Key
            Constants.USER_ID, // User ID
            (result, error) -> {
                if (error != null) {
                    Log.e(TAG, "SDK initialization failed", error);
                    
                    // Update status to show initialization failed
                    runOnUiThread(() -> {
                        binding.statusLabel.setText("SDK Status: ❌ Failed to Initialize");
                        binding.statusLabel.setTextColor(ContextCompat.getColor(MainActivity.this, android.R.color.holo_red_dark));
                    });
                } else {
                    Log.d(TAG, "SDK initialized successfully: " + result);
                    
                    // Enable buttons and update status after successful initialization
                    runOnUiThread(() -> {
                        binding.statusLabel.setText("SDK Status: ✅ Initialized");
                        binding.statusLabel.setTextColor(ContextCompat.getColor(MainActivity.this, android.R.color.holo_green_dark));
                        binding.predictButton.setEnabled(true);
                        binding.logEventButton.setEnabled(true);
                        
                         // Load fake products after initialization for logging
                        loadFakeProducts();
                    });
                    
                    // Set discount info change callback AFTER initialization
                    // This should now work regardless of timing
                    MonetaiSDKJava.getShared().setOnDiscountInfoChange(discount -> {
                        Log.d(TAG, "Discount info changed: " + discount);
                        handleDiscountInfoChange(discount);
                    });
                    
                    // Set up paywall AFTER initialization
                    setupPaywall();
                }
            }
        );
    }
    
    
    private void handleDiscountInfoChange(AppUserDiscount discountInfo) {
        runOnUiThread(() -> {
            if (discountInfo != null) {
                Date now = new Date();
                Date endTime = discountInfo.getEndedAt();

                // Debug time comparison
                System.out.println("🔍 Time Debug:");
                System.out.println("  Current time (local): " + now);
                System.out.println("  End time (from API): " + endTime);
                System.out.println("  Time comparison: now < endTime = " + (now.before(endTime)));
                System.out.println("  Time difference (ms): " + (endTime.getTime() - now.getTime()));

                if (now.before(endTime)) {
                    // Discount is valid - SDK banner will be shown automatically
                    binding.discountStatusLabel.setText("Discount: ✅ Active (Expires: " + dateFormat.format(endTime) + ")");
                    binding.discountStatusLabel.setTextColor(ContextCompat.getColor(this, android.R.color.holo_green_dark));
                } else {
                    // Discount expired
                    binding.discountStatusLabel.setText("Discount: ❌ Expired");
                    binding.discountStatusLabel.setTextColor(ContextCompat.getColor(this, android.R.color.holo_red_dark));
                }
            } else {
                // No discount
                binding.discountStatusLabel.setText("Discount: None");
                binding.discountStatusLabel.setTextColor(ContextCompat.getColor(this, android.R.color.darker_gray));
            }
        });
    }


    // MARK: - Button Actions
    private void predictButtonTapped() {
        executor.execute(() -> {
            try {
                MonetaiSDKJava.getShared().predict((result, error) -> {
                    if (error != null) {
                        System.out.println("Prediction failed: " + error);
                        runOnUiThread(() -> {
                            binding.resultLabel.setText("❌ Prediction failed: " + error.getMessage());
                            binding.resultLabel.setTextColor(ContextCompat.getColor(MainActivity.this, android.R.color.holo_red_dark));
                        });
                    } else {
                        // Check if result is null before accessing its methods
                        if (result != null) {
                            System.out.println("Prediction result: " + result.getPrediction());
                            System.out.println("Test group: " + result.getTestGroup());

                            if (result.getPrediction() != null) {
                                PredictResult prediction = result.getPrediction();
                                if (prediction == PredictResult.NON_PURCHASER) {
                                    // When predicted as non-purchaser, SDK should automatically show paywall
                                    System.out.println("Predicted as non-purchaser - SDK should automatically show paywall");
                                    System.out.println("Note: If paywall doesn't appear, check SDK's automatic display logic");
                                } else if (prediction == PredictResult.PURCHASER) {
                                    // When predicted as purchaser
                                    System.out.println("Predicted as purchaser - paywall not needed");
                                }
                            } else {
                                // When prediction is null
                                System.out.println("No prediction available");
                            }

                            runOnUiThread(() -> {
                                binding.resultLabel.setText("✅ Prediction completed - check console for details");
                                binding.resultLabel.setTextColor(ContextCompat.getColor(MainActivity.this, android.R.color.holo_green_dark));

                                // Show alert with prediction result
                                new AlertDialog.Builder(MainActivity.this)
                                    .setTitle("Purchase Prediction")
                                    .setMessage("Prediction: " + result.getPrediction() + "\nTest Group: " + result.getTestGroup())
                                    .setPositiveButton("OK", null)
                                    .show();
                            });
                        } else {
                            System.out.println("Prediction result is null");
                            runOnUiThread(() -> {
                                binding.resultLabel.setText("❌ Prediction result is null");
                                binding.resultLabel.setTextColor(ContextCompat.getColor(MainActivity.this, android.R.color.holo_red_dark));
                            });
                        }
                    }
                });

            } catch (Exception error) {
                runOnUiThread(() -> {
                    binding.resultLabel.setText("❌ Prediction failed: " + error.getMessage());
                    binding.resultLabel.setTextColor(ContextCompat.getColor(this, android.R.color.holo_red_dark));
                });
                System.out.println("Prediction failed: " + error);
            }
        });
    }

    private void logEventButtonTapped() {
        executor.execute(() -> {
            // Log a sample event with parameters
            Map<String, Object> params = new HashMap<>();
            params.put("button", "test_button");
            params.put("screen", "main");
            
            MonetaiSDKJava.getShared().logEvent("button_click", params);

            runOnUiThread(() -> {
                binding.resultLabel.setText("✅ Event logged: button_click\nParameters: button=test_button, screen=main");
                binding.resultLabel.setTextColor(ContextCompat.getColor(this, android.R.color.holo_green_dark));
            });

            System.out.println("Event logged: button_click");
        });
    }
    
    // MARK: - Fake Products
    private void loadFakeProducts() {
        fakeProductAdapter.submitList(fakeProducts);
    }
    
    private void logFakeProductViewed(FakeProduct product) {
        if (!MonetaiSDKJava.getInitialized()) return;
        
        ViewProductItemParams params = new ViewProductItemParams(
            product.id,
            product.price,
            product.regularPrice,
            product.currencyCode,
            product.month
        );
        
        executor.execute(() -> {
            try {
                MonetaiSDKJava.getShared().logViewProductItem(params);
                Log.d(TAG, "✅ Logged fake product view: " + product.id);
            } catch (Exception e) {
                Log.e(TAG, "Failed to log fake product view: " + product.id, e);
            }
        });
    }
    
    // MARK: - Paywall Methods
    private void setupPaywall() {
        paywallConfig = MonetaiSDKJava.createPaywallConfigBuilder()
            .discountPercent(30)
            .regularPrice("$99.99")
            .discountedPrice("$69.99")
            .locale("en")
            .style(PaywallStyle.COMPACT)
            .features(Arrays.asList(
                new Feature("Unlimited Access", "Use all premium features without limits", true),
                new Feature("Advanced Analytics", "AI-powered insights", false),
                new Feature("Priority Support", "24/7 customer support", false)
            ))
            .enabled(true)
            .bannerBottom(20f)
            .isSubscriber(false)
            .onPurchase(new MonetaiSDKJava.OnPurchaseCallback() {
                @Override
                public void onPurchase(com.monetai.sdk.models.PaywallContext paywallContext, Runnable closePaywall) {
                    System.out.println("🛒 Purchase button tapped in paywall!");
                    System.out.println("  📱 Activity: " + paywallContext.getActivity().getClass().getSimpleName());
                    System.out.println("  🏗️ App Context: " + paywallContext.getApplicationContext().getClass().getSimpleName());
                    
                    runOnUiThread(() -> {
                        // Simulate subscription purchase
                        MonetaiSDKJava.setSubscriptionStatus(true);
                        
                        // Show success toast
                        Toast.makeText(MainActivity.this, "🎉 Purchase successful! Welcome to Premium!", Toast.LENGTH_SHORT).show();
                    });
                    
                    // Close paywall
                    closePaywall.run();
                }
            })
            .onTermsOfService(new MonetaiSDKJava.OnTermsOfServiceCallback() {
                @Override
                public void onTermsOfService(com.monetai.sdk.models.PaywallContext paywallContext) {
                    System.out.println("📄 Terms of Service tapped!");
                    System.out.println("  📱 Activity: " + paywallContext.getActivity().getClass().getSimpleName());
                    showTermsOfService(paywallContext);
                }
            })
            .onPrivacyPolicy(new MonetaiSDKJava.OnPrivacyPolicyCallback() {
                @Override
                public void onPrivacyPolicy(com.monetai.sdk.models.PaywallContext paywallContext) {
                    System.out.println("🔒 Privacy Policy tapped!");
                    System.out.println("  📱 Activity: " + paywallContext.getActivity().getClass().getSimpleName());
                    showPrivacyPolicy(paywallContext);
                }
            })
            .build();
        
        if (paywallConfig != null) {
            // Configure paywall
            MonetaiSDKJava.configurePaywall(paywallConfig);
            System.out.println("🎯 Paywall configured successfully");
        }
    }
    
    // MARK: - Paywall Callback Helpers
    private static void showTermsOfService(com.monetai.sdk.models.PaywallContext paywallContext) {
        System.out.println("📄 Showing Terms of Service...");
        System.out.println("  🎯 Context from: " + paywallContext.getActivity().getClass().getSimpleName());
        
        // Use the Activity from PaywallContext for better dialog management
        android.app.Activity contextActivity = paywallContext.getActivity();
        
        // PaywallContext의 Activity는 이미 MonetaiPaywallActivity이므로 직접 사용
        contextActivity.runOnUiThread(() -> {
            String contextInfo = "Called from: " + paywallContext.getActivity().getClass().getSimpleName();
            
            // 가장 간단한 기본 다이얼로그 - PaywallActivity에서 직접 표시
            new AlertDialog.Builder(contextActivity)
                .setTitle("Terms of Service")
                .setMessage("Terms of Service content goes here...\n\n" + contextInfo)
                .setPositiveButton("Accept", null)
                .setNegativeButton("Cancel", null)
                .show();
            
            System.out.println("✅ Terms of Service dialog shown on PaywallActivity: " + contextActivity.getClass().getSimpleName());
        });
    }
    
    private static void showPrivacyPolicy(com.monetai.sdk.models.PaywallContext paywallContext) {
        System.out.println("🔒 Showing Privacy Policy...");
        System.out.println("  🎯 Context from: " + paywallContext.getActivity().getClass().getSimpleName());
        
        // Use the Activity from PaywallContext for better dialog management
        android.app.Activity contextActivity = paywallContext.getActivity();
        
        // PaywallContext의 Activity는 이미 MonetaiPaywallActivity이므로 직접 사용
        contextActivity.runOnUiThread(() -> {
            String contextInfo = "Called from: " + paywallContext.getActivity().getClass().getSimpleName();
            
            // 가장 간단한 기본 다이얼로그 - PaywallActivity에서 직접 표시
            new AlertDialog.Builder(contextActivity)
                .setTitle("Privacy Policy")
                .setMessage("Privacy Policy content goes here...\n\n" + contextInfo)
                .setPositiveButton("Accept", null)
                .setNegativeButton("Cancel", null)
                .show();
            
            System.out.println("✅ Privacy Policy dialog shown on PaywallActivity: " + contextActivity.getClass().getSimpleName());
        });
    }
} 

File: examples/sample-kotlin/src/main/java/com/monetai/sample/kotlin/MainActivity.kt
package com.monetai.sample.kotlin

import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import com.monetai.sample.kotlin.databinding.ActivityMainBinding
import com.monetai.sdk.MonetaiSDK
import com.monetai.sdk.models.ViewProductItemParams
import com.monetai.sdk.models.PredictResult
import com.monetai.sdk.models.Feature
import com.monetai.sdk.models.PaywallConfig
import com.monetai.sdk.models.PaywallStyle
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : AppCompatActivity() {

    // MARK: - UI Elements
    private lateinit var binding: ActivityMainBinding
    
    // MARK: - Paywall State
    private var isSubscriber: Boolean = false
    private var paywallConfig: PaywallConfig? = null
    
    // MARK: - Fake Products
    private lateinit var fakeProductAdapter: FakeProductAdapter
    private val fakeProducts: List<FakeProduct> = listOf(
        FakeProduct(
            id = "fake_basic_monthly",
            title = "Basic Plan",
            description = "Essential features for starters",
            price = 4.99,
            regularPrice = 9.98,
            currencyCode = "USD",
            month = 1
        ),
        FakeProduct(
            id = "fake_pro_yearly",
            title = "Pro Plan",
            description = "Advanced tools for power users",
            price = 59.99,
            regularPrice = 119.98,
            currencyCode = "USD",
            month = 12
        )
    )

    // MARK: - Constants
    companion object {
        private val dateFormat = SimpleDateFormat("MMM dd, HH:mm", Locale.getDefault())
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        setupUI()
        
        // Log app launch event BEFORE SDK initialization
        MonetaiSDK.shared.logEvent("app_in")
        
        setupMonetaiSDK()
        
        // Initialize subscription UI
        updateSubscriptionUI()
    }

    // MARK: - UI Setup
    private fun setupUI() {
        // Set initial text
        binding.titleLabel.text = "MonetaiSDK Demo"
        binding.statusLabel.text = "SDK Status: Initializing..."
        binding.discountStatusLabel.text = "Discount: None"


        // Setup button click listeners
        binding.predictButton.setOnClickListener { predictButtonTapped() }
        binding.logEventButton.setOnClickListener { logEventButtonTapped() }

        // Disable buttons initially
        binding.predictButton.isEnabled = false
        binding.logEventButton.isEnabled = false
        binding.predictButton.alpha = 0.5f
        binding.logEventButton.alpha = 0.5f
        
        // Fake products list
        fakeProductAdapter = FakeProductAdapter { product ->
            logFakeProductViewed(product)
        }
        binding.recyclerViewFakeProducts.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = fakeProductAdapter
        }
    }

    // MARK: - MonetaiSDK Setup
    private fun setupMonetaiSDK() {
        // Initialize SDK FIRST
        MonetaiSDK.shared.initialize(
            context = this@MainActivity,
            sdkKey = Constants.SDK_KEY,
            userId = Constants.USER_ID
        ) { result, error ->
            runOnUiThread {
                if (error != null) {
                    // SDK initialization failed
                    updateSDKStatus()
                } else {
                    // SDK initialized successfully
                    updateSDKStatus()
                    
                    // Load fake products after initialization for logging
                    loadFakeProducts()
                    
                    // Set up discount info change callback AFTER initialization
                    MonetaiSDK.shared.onDiscountInfoChange = { discountInfo ->
                        runOnUiThread {
                            println("🎯 Discount info changed callback triggered")
                            println("  discountInfo: $discountInfo")
                            println("  current time: ${Date()}")
                            if (discountInfo != null) {
                                println("  discount end time: ${discountInfo.endedAt}")
                                println("  is discount active: ${discountInfo.endedAt > Date()}")
                            }
                            handleDiscountInfoChange(discountInfo)
                        }
                    }
                    
                    // Set up paywall AFTER initialization
                    setupPaywall()
                }
            }
        }
    }

    private fun updateSDKStatus() {
        val isInitialized = MonetaiSDK.shared.getInitialized()
        binding.statusLabel.text = "SDK Status: ${if (isInitialized) "✅ Initialized" else "❌ Not Initialized"}"
        
        // Enable buttons only after initialization
        binding.predictButton.isEnabled = isInitialized
        binding.logEventButton.isEnabled = isInitialized
        binding.predictButton.alpha = if (isInitialized) 1.0f else 0.5f
        binding.logEventButton.alpha = if (isInitialized) 1.0f else 0.5f
    }

    private fun handleDiscountInfoChange(discountInfo: com.monetai.sdk.models.AppUserDiscount?) {
        if (discountInfo != null) {
            val now = Date()
            val endTime = discountInfo.endedAt

            // Debug time comparison
            println("🔍 Time Debug:")
            println("  Current time (local): $now")
            println("  End time (from API): $endTime")
            println("  Time comparison: now < endTime = ${now < endTime}")
            println("  Time difference (ms): ${endTime.time - now.time}")

            if (now < endTime) {
                // Discount is valid - SDK will automatically show banner
                binding.discountStatusLabel.text = "Discount: ✅ Active (Expires: ${dateFormat.format(endTime)})"
                binding.discountStatusLabel.setTextColor(ContextCompat.getColor(this, android.R.color.holo_green_dark))
                println("🎯 Discount is active - SDK should automatically show banner")
            } else {
                // Discount expired
                binding.discountStatusLabel.text = "Discount: ❌ Expired"
                binding.discountStatusLabel.setTextColor(ContextCompat.getColor(this, android.R.color.holo_red_dark))
                println("❌ Discount expired - SDK should automatically hide banner")
            }
        } else {
            // No discount
            binding.discountStatusLabel.text = "Discount: None"
            binding.discountStatusLabel.setTextColor(ContextCompat.getColor(this, android.R.color.darker_gray))
            println("📭 No discount - SDK should automatically hide banner")
        }
    }



    // MARK: - Button Actions
    private fun predictButtonTapped() {
        MonetaiSDK.shared.predict { result, error ->
            runOnUiThread {
                if (error != null) {
                    println("Prediction failed: $error")
                } else {
                    result?.let { predictResult ->
                        println("Prediction result: ${predictResult.prediction}")
                        println("Test group: ${predictResult.testGroup}")

                        when (predictResult.prediction) {
                            PredictResult.NON_PURCHASER -> {
                                // When predicted as non-purchaser, SDK should automatically show paywall
                                println("Predicted as non-purchaser - SDK should automatically show paywall")
                                println("Note: If paywall doesn't appear, check SDK's automatic display logic")
                            }
                            PredictResult.PURCHASER -> {
                                // When predicted as purchaser
                                println("Predicted as purchaser - paywall not needed")
                            }
                            null -> {
                                // When prediction is null
                                println("No prediction available")
                            }
                        }

                        // Show alert with prediction result
                        AlertDialog.Builder(this@MainActivity)
                            .setTitle("AI Purchase Prediction Result")
                            .setMessage("Prediction: ${predictResult.prediction}\nTest Group: ${predictResult.testGroup}")
                            .setPositiveButton("OK", null)
                            .show()
                    }
                }
            }
        }
    }
    
    // MARK: - Paywall Methods
    private fun setupPaywall() {
        paywallConfig = PaywallConfig(
            discountPercent = 30,
            regularPrice = "$99.99",
            discountedPrice = "$69.99",
            locale = "ko",
            style = PaywallStyle.COMPACT,
            features = listOf(
                Feature(
                    title = "Unlimited Access",
                    description = "Use all premium features without limits",
                    isPremiumOnly = true
                ),
                Feature(
                    title = "Advanced Analytics",
                    description = "AI-powered insights",
                    isPremiumOnly = false
                ),
                Feature(
                    title = "Priority Support",
                    description = "24/7 customer support",
                    isPremiumOnly = false
                )
            ),
            enabled = true,
            bannerBottom = 20f,
            isSubscriber = isSubscriber,
            
            
            onPurchase = { paywallContext, closePaywall ->
                println("💰 MainActivity: onPurchase callback triggered")
                println("  📱 Activity: ${paywallContext.activity.javaClass.simpleName}")
                println("  🏗️ App Context: ${paywallContext.applicationContext.javaClass.simpleName}")
                
                // Process purchase immediately
                println("💰 Processing purchase...")
                // Update subscription status
                isSubscriber = true
                MonetaiSDK.shared.setSubscriptionStatus(true)
                updateSubscriptionUI()
                // Close paywall
                closePaywall()
                println("✅ Purchase completed successfully")
            },
            onTermsOfService = { paywallContext ->
                println("🔗 MainActivity: onTermsOfService callback triggered")
                println("  📱 Activity: ${paywallContext.activity.javaClass.simpleName}")
                showTermsOfService(paywallContext)
            },
            onPrivacyPolicy = { paywallContext ->
                println("🔗 MainActivity: onPrivacyPolicy callback triggered") 
                println("  📱 Activity: ${paywallContext.activity.javaClass.simpleName}")
                showPrivacyPolicy(paywallContext)
            }
        )
        
        paywallConfig?.let { config ->
            // Configure paywall
            MonetaiSDK.shared.configurePaywall(config)
            println("🎯 Paywall configured successfully")
            println("  config.enabled: ${config.enabled}")
            println("  config.isSubscriber: ${config.isSubscriber}")
            println("  config.bannerBottom: ${config.bannerBottom}")
        }
    }
    
    private fun updateSubscriptionUI() {
        val statusText = if (isSubscriber) "✅ Subscriber" else "❌ Non-subscriber"
        binding.subscriptionStatusLabel.text = "Subscription Status: $statusText"
        binding.subscriptionStatusLabel.setTextColor(
            if (isSubscriber) 
                ContextCompat.getColor(this, android.R.color.holo_green_dark)
            else 
                ContextCompat.getColor(this, android.R.color.holo_red_dark)
        )
    }
    
    
        private fun showTermsOfService(paywallContext: com.monetai.sdk.models.PaywallContext) {
        println("📄 MainActivity: showTermsOfService() called")
        println("  🎯 Context from: ${paywallContext.activity.javaClass.simpleName}")
        
        // Use the Activity from PaywallContext for better dialog management
        val contextActivity = paywallContext.activity
        
        // PaywallContext의 Activity는 이미 MonetaiPaywallActivity이므로 직접 사용
        contextActivity.runOnUiThread {
            val contextInfo = "Called from: ${paywallContext.activity.javaClass.simpleName}"
            
            // 가장 간단한 기본 다이얼로그 - PaywallActivity에서 직접 표시
            AlertDialog.Builder(contextActivity)
                .setTitle("Terms of Service")
                .setMessage("Terms of service content will be displayed here.\n\n$contextInfo")
                .setPositiveButton("OK", null)
                .show()
            
            println("✅ Terms of Service dialog shown on PaywallActivity: ${contextActivity.javaClass.simpleName}")
        }
    }
    
        private fun showPrivacyPolicy(paywallContext: com.monetai.sdk.models.PaywallContext) {
        println("🔒 MainActivity: showPrivacyPolicy() called")
        println("  🎯 Context from: ${paywallContext.activity.javaClass.simpleName}")
        
        // Use the Activity from PaywallContext for better dialog management
        val contextActivity = paywallContext.activity
        
        // PaywallContext의 Activity는 이미 MonetaiPaywallActivity이므로 직접 사용
        contextActivity.runOnUiThread {
            val contextInfo = "Called from: ${paywallContext.activity.javaClass.simpleName}"
            
            // 가장 간단한 기본 다이얼로그 - PaywallActivity에서 직접 표시
            AlertDialog.Builder(contextActivity)
                .setTitle("Privacy Policy")
                .setMessage("Privacy policy content will be displayed here.\n\n$contextInfo")
                .setPositiveButton("OK", null)
                .show()
            
            println("✅ Privacy Policy dialog shown on PaywallActivity: ${contextActivity.javaClass.simpleName}")
        }
    }

    private fun logEventButtonTapped() {
        // Log a sample event with parameters
        val params = mapOf(
            "button" to "test_button",
            "screen" to "main"
        )
        
        MonetaiSDK.shared.logEvent("button_click", params)



        println("Event logged: button_click")
    }
    
    // MARK: - Fake Products
    private fun loadFakeProducts() {
        fakeProductAdapter.submitList(fakeProducts)
    }
    
    private fun logFakeProductViewed(product: FakeProduct) {
        if (!MonetaiSDK.shared.getInitialized()) return
        
        val params = ViewProductItemParams(
            productId = product.id,
            price = product.price,
            regularPrice = product.regularPrice,
            currencyCode = product.currencyCode,
            month = product.month
        )
        
        try {
            MonetaiSDK.shared.logViewProductItem(params)
            println("✅ Logged fake product view: ${product.id}")
        } catch (e: Exception) {
            println("❌ Failed to log fake product view: ${product.id}, error: $e")
        }
    }
} 